devito/finite_differences/tools.py

```diff
@@ -261,3 +261,22 @@ def generate_indices_staggered(func, dim, order, side=None, x0=None):
             ind = [start, start - diff]
 
     return start, tuple(ind)
+
+
+def make_get_shifted_oringin(shift, shape):
```

Comment:
there's a typo -- it'd be "origin" not "origin"

since you assign the function to a variable named "shift_x0", I'd call this function "make_shift_x0"